velox/functions/prestosql/aggregates/AverageAggregate.cpp

```diff
@@ -134,7 +143,12 @@ class AverageAggregate : public exec::Aggregate {
       if (!decodedRaw_.isNullAt(0)) {
         const TInput value = decodedRaw_.valueAt<TInput>(0);
         const auto numRows = rows.countSelected();
-        updateNonNullValue(group, numRows, TAccumulator(value) * numRows);
+        if constexpr (aggregateType == AggregateType::kAvg) {
```

Comment:
It might look cleaner if value above is not const and then
if (constexpr (aggregateType == AggregateType::kGeoMean) {
  value = log10(value);
}
updateNonNullValue(...)